File: code/02-Data-Engineering/scala/06-BatchJob/Workflow.scala
// Databricks notebook source
// MAGIC %run ./GlobalVarsAndMethods

// COMMAND ----------

val batchID: Int = generateBatchID()

// COMMAND ----------

insertBatchMetadata(batchID,1,"Execute report 1","Started")
val executionStatusReport1 = dbutils.notebook.run("Report-1", 60)

if(executionStatusReport1 == "Pass")
  insertBatchMetadata(batchID,1,"Execute report 1","Completed")

// COMMAND ----------

var executionStatusReport2 = "-"
if(executionStatusReport1 == "Pass")
{
  insertBatchMetadata(batchID,2,"Execute report 2","Started")
  executionStatusReport2 = dbutils.notebook.run("Report-2", 60)
  insertBatchMetadata(batchID,2,"Execute report 2","Completed")
  
}

// COMMAND ----------

dbutils.notebook.exit(executionStatusReport2)

File: code/02-Data-Engineering/pyspark/06-BatchJob/Workflow.py
# Databricks notebook source
# MAGIC %run ./GlobalVarsAndMethods

# COMMAND ----------

insertBatchMetadata(1,"Execute report 1","Started")
executionStatusReport1 = dbutils.notebook.run("Report-1", 60)

if(executionStatusReport1 == "Pass"):
  insertBatchMetadata(1,"Execute report 1","Completed")
else:
  insertBatchMetadata(1,"Execute report 1","Failed")

# COMMAND ----------

executionStatusReport2 = "-"
if(executionStatusReport1 == "Pass"):
  insertBatchMetadata(2,"Execute report 2","Started")
  executionStatusReport2 = dbutils.notebook.run("Report-2", 60)
  if(executionStatusReport2 == "Pass"):
    insertBatchMetadata(2,"Execute report 2","Completed")
  else:
    insertBatchMetadata(2,"Execute report 2","Failed")  


# COMMAND ----------

if(executionStatusReport1 == "Pass"):
  dbutils.notebook.exit(executionStatusReport2)
else:
  dbutils.notebook.exit(executionStatusReport1)